tests/test_coreg/test_workflows.py

```diff
@@ -256,9 +256,58 @@ def test_dem_coregistration(self) -> None:
         out_fig.close()
 
         # Testing different coreg method
-        dem_coreg, coreg_method, coreg_stats, inlier_mask = dem_coregistration(
+        dem_coreg2, coreg_method2, coreg_stats2, inlier_mask2 = dem_coregistration(
             tba_dem, ref_dem, coreg_method=xdem.coreg.Deramp()
         )
-        assert isinstance(coreg_method, xdem.coreg.Deramp)
-        assert abs(coreg_stats["med_orig"].values) > abs(coreg_stats["med_coreg"].values)
-        assert coreg_stats["nmad_orig"].values > coreg_stats["nmad_coreg"].values
+        assert isinstance(coreg_method2, xdem.coreg.Deramp)
+        assert abs(coreg_stats2["med_orig"].values) > abs(coreg_stats2["med_coreg"].values)
+        assert coreg_stats2["nmad_orig"].values > coreg_stats2["nmad_coreg"].values
+
+        # Testing with initial shift
+        test_shift_list = [10, 5]
+        tba_dem_shift = tba_dem.copy()
+        dem_coreg2, coreg_method2, coreg_stats2, inlier_mask2 = dem_coregistration(
+            tba_dem_shift, ref_dem, estimated_initial_shift=test_shift_list
+        )
+        tba_dem_shift_test = tba_dem.translate(test_shift_list[0] * tba_dem.res[0], test_shift_list[1] * tba_dem.res[1])
+        assert tba_dem_shift.raster_equal(tba_dem_shift_test)
```

Comment:
Reading this test (and looking again at the code in `workflows`), I realize that the input `tba_dem` is shifted in-place, which thus modifies the input from the user.
Until now our practice has been to "never authorize the modification of any user input", unless the function has a clear `inplace` argument (when returning same type as self) or is called `set_xxx` (when re-setting an attribute).

If that makes sense to you as well, we should return a copy in the `workflows` function instead of modifying the input, I'll tag where :slightly_smiling_face: